src/Runner.Listener/Runner.cs

```diff
@@ -309,6 +309,15 @@ public async Task<int> ExecuteCommand(CommandSettings command)
                         _term.WriteLine("https://docs.github.com/en/actions/hosting-your-own-runners/autoscaling-with-self-hosted-runners#using-ephemeral-runners-for-autoscaling", ConsoleColor.Yellow);
                     }
 
+                    var cred = store.GetCredentials();
+                    if (cred != null &&
+                        cred.Scheme == Constants.Configuration.OAuth &&
+                        cred.Data.ContainsKey("EnableAuthMigrationByDefault"))
```

Comment:
`EnableAuthMigrationByDefault` in the `.credentials` will make the runner to try migration without waiting for service.

Reply:
it should never be null, however, i added check since this a new code path, and i can't patch the runner after it releases.